File: Learn/Algorithms/Warmup/Diagonal Difference/diagonaldiff.c
#include <stdio.h>

int main() {
    int n;

    scanf("%d", &n);

    int values[n][n];

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            scanf("%d", &(values[i][j]));
        }
    }
    int first = 0;
    int second = 0;

    for (int i = 0; i < n; i++) {
        first += values[i][i];
        second += values[i][(n - 1) - i];
    }

    int diff = first - second;

    // Get absolute values
    if (diff < 0) {
        diff *= -1;
    }

    printf("%d\n", diff);

    return 0;

}

File: Learn/Algorithms/Warmup/Diagonal Difference/diagonaldiff.py
# Supplied code
n = int(input().strip())
a = []
for a_i in range(n):
    a_t = [int(a_temp) for a_temp in input().strip().split(' ')]
    a.append(a_t)

# Start
first = 0
second = 0
for i in range(n):
    first += a[i][i]
    second += a[i][(n - 1) - i]

diff = first - second

if diff < 0:
    diff *= -1

print(diff)
